src/PgRvn.Server/PowerBI/PBIPreviewQuery.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Raven.Client.Documents;
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PgRvn.Server
+{
+    public class PBIPreviewQuery : RqlQuery
+    {
+        public string _tableName;
+        private List<PgDataRow> _results;
+        private static readonly Regex _regex = new Regex(@"(?is) ^\s* select\s +.*\s + from\s + INFORMATION_SCHEMA.columns\s + where\s + TABLE_SCHEMA\s +=\s + 'public'\s + and\s + TABLE_NAME\s *=\s * '(?<table_name>[^']+)'\s+order\s+by\s+TABLE_SCHEMA\s*,\s*TABLE_NAME\s*,\s*ORDINAL_POSITION\s*$",
+            RegexOptions.Compiled);
```

Comment:
what are you matching here?

Reply:
Added documentation - we're matching the PowerBI query that this class is responsible for handling.